src/Listener/ToHeader.php

```diff
@@ -70,11 +70,12 @@ protected function getContactNames($tasks) {
       $ids[$task->getContactId()] = $task->getContactId();
     }
 
+    $ids = array_filter($ids, 'is_numeric');
     if (!$ids) {
```

Comment:
@jitendrapurohit 
should we do  `if (!empty($ids)) {` instead?

Reply:
Thanks @jitendrapurohit! I can confirm that patch resolves the crash when opening `civicrm/mailing/view` anonymously. Nice.

I also share @yashodha's sense that `empty()` is better than implicitly casting to a boolean. I'm not really sure why, and it fixes either way, but it's easy to tack on an extra commit for that.

Rebased+merged.